src/cluster/managers/AnnouncementManager.ts

```diff
@@ -0,0 +1,108 @@
+
+        let channel = await this.#util.getChannel(config.channel);
+        const role = await this.#util.getRole(guild.id, config.role);
+
+        if (channel === undefined || !guard.isGuildChannel(channel) || !guard.isTextableChannel(channel) || guard.isThreadChannel(channel) && channel.threadMetadata.archived)
+            channel = undefined;
+
+        return { channel, role };
+    }
+
+    public async loadConfig(guild: Guild, user: User, queryChannel: KnownTextableChannel, options?: Partial<AnnouncementConfigOptions>): Promise<AnnounceResult> {
+        const command = await this.#commands.get('announce', guild, user);
+        if (command.state !== 'ALLOWED')
+            return { state: 'NotAllowed' };
+
+        let { channel, role } = options ?? await this.getCurrentConfig(guild);
+        if (channel === undefined) {
+            const result = await this.#util.queryChannel({
+                actors: [user],
+                context: queryChannel,
+                choices: guild.channels.filter(guard.isTextableChannel).values(),
+                prompt: 'ℹ️ Please select the channel that announcements should be put in.'
+            });
+
+            if (result.state === 'TIMED_OUT' || result.state === 'CANCELLED')
+                return { state: 'TimedOut' };
+
+            if (result.state === 'SUCCESS')
+                channel = result.value;
+        }
+
+        if (channel === undefined)
+            return { state: 'ChannelNotFound' };
+        if (!guard.isGuildChannel(channel) || channel.guild !== guild)
+            return { state: 'ChannelNotInGuild' };
+        if (!guard.isTextableChannel(channel))
+            return { state: 'ChannelInvalid' };
+
+        if (role === undefined) {
+            const result = await this.#util.queryRole({
+                actors: [user],
+                context: queryChannel,
+                prompt: 'ℹ️ Please select the role to mention when announcing.',
+                guild: guild
+            });
+
+            if (result.state === 'TIMED_OUT' || result.state === 'CANCELLED')
+                return { state: 'TimedOut' };
+
+            if (result.state === 'SUCCESS')
+                role = result.value;
+        }
+        if (role === undefined)
+            return { state: 'RoleNotFound' };
+
+        await this.#database.setAnnouncements(guild.id, {
+            channel: channel.id,
+            role: role.id
+        });
+        return { state: 'Success', detail: { channel, role } };
+    }
+}
+
+interface AnnouncementConfig {
+    readonly channel: KnownGuildTextableChannel;
+    readonly role: Role;
+}
+
+interface AnnouncementConfigOptions {
+    readonly channel?: KnownChannel;
+    readonly role?: Role;
+}
+
+type AnnounceResult =
+    | Result<'Success', AnnouncementConfig>
+    | Result<'NotAllowed'>
+    | Result<'ChannelNotFound'>
+    | Result<'ChannelNotInGuild'>
+    | Result<'ChannelInvalid'>
+    | Result<'RoleNotFound'>
+    | Result<'TimedOut'>
```

Comment:
While I understand those are only used here, I am not sure we follow the same conventions regarding other managers. How can we refactor our code to fit a standardized way of making managers?

Reply:
I think having it like this is better, the types should be placed where they are used. The `types.ts` files arent the best way to do things imo, but when the entire solution is being rewritten, it was definitely the easiest. Imo the only time types should be placed in a `types.ts` going forwards is when they are an abstract concept, like the `Result` type being used here.